### ISSUE TITLE
Fun compilation server static var leak 

### ISSUE THREAD

**Issue by RblSb** [HaxeFoundation/haxe#11817]
Create two separate files Main.hx/Utils.hx in base haxe project:
```haxe
class Main {
	static function main() {
		// Utils.foo;
		// Utils.foo_leak;
	}
}

class Utils {
	public static function foo():Int {
		static var leak = 0;
		return leak;
	}
}
```

Reload window, uncomment `Utils.foo;` line in `main` and then `Utils.foo_leak;`, so it will magically show in completion and compile. Does not work in single module or if you uncomment `foo_leak` before `foo`.
Instead of uncommenting, you can type both calls in same order to see `leak` after `Utils.foo;` in completion.

[staticsanity.zip](https://github.com/user-attachments/files/17693501/staticsanity.zip)

**Comment by Simn** [comment]
Note that this isn't purely a completion issue, compilation should also not succeed.

It's fine for the field to exist in the cache, so we'll have to restrict explicit access to it.

**Comment by Simn** [comment]
Closed by #11825
